doc/adrs/128_add_support_for_confirmation_dialogs.adoc

```diff
@@ -0,0 +1,53 @@
+= ADR-128 - Add support for confirmation dialogs
+
+== Context
+
+Sirius Web does not have an undo/redo mechanism.
+When an element has been deleted, either from the Explorer view, from any representation, or any other view, there is no way to retrieve it.
+All delete actions are executable in one click.
+Any accidental click on a delete action can lead to the unwanted loss of data.
+By extension, some one-click actions may lead to unwanted consequences.
+
+== Decision
+
+The first decision is about the delete actions (click on the delete tool on a node, click on DEL key on a node, click on the delete menu item on an object in the Explorer view).
+For these actions, a confirmation dialog will be added, asking to confirm the deletion.
+It will be possible to disable these dialogs by clicking on the "Don't ask me again" checkbox inside them.
+These checkboxes will be associated to cookies, allowing to remember which dialogs have been disabled or not, for each end-users.
+
+A new `Confirmation` concept will be added into `Tool` interface in `diagram.graphqls`:
+
+```
+interface Tool {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+  confirmation: Confirmation!
+}
+
+interface Confirmation {
+  id: ID!
+  enable: Boolean! // if the confirmation dialog (or any other graphical element) should be enabled on this tool
+  message: String! // the message that will be displayed on confirmation
+  label: String! // the label that will be displayed on the confirmation button (or any other graphical element used to confirm the execution of the tool)
+}
+```
+
+On frontend side, a new react component will be created for the confirmation dialog.
+```
+<ConfirmationDialog message={message} label={label} onConfirm={// execution of the tool}/>
```

Comment:
I think you would need to have more than that, for example the id of the confirmation dialog to know which dialog to ignore (you would probably end up with `GQLConfirmationDialog` directly as a props).

It does not matter right now in this ADR but since you will end up using this dialog at multiple location with its internal lifecycle (to open it or not and then running the confirmation directly instead depending on the local storage state), you may end up with a quite different API than this).

The current example visible here is fine for an ADR, no need to change anything.